bukkit/src/main/java/net/william278/husktowns/events/PlayerEnterTownEvent.java

```diff
@@ -0,0 +1,64 @@
+package net.william278.husktowns.events;
+
+import net.william278.husktowns.town.Town;
+import org.bukkit.Location;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.player.PlayerEvent;
+import org.bukkit.event.player.PlayerMoveEvent;
+import org.jetbrains.annotations.NotNull;
+
+public class PlayerEnterTownEvent extends PlayerEvent implements Cancellable {
+
+    private static final HandlerList HANDLER_LIST = new HandlerList();
+    private boolean isCancelled;
+
+    private final String enteredTown; // You should make ClaimedChunk#getTown() return a Town object instead of a String
```

Comment:
Yeah, I should. The reason for this was deliberate due to the design choices I made in terms of making a separation between _data stored in the cache_ versus _data stored in the database_; the latter of which would return the full town object API. Thus, the API makes some weird choices in how I originally exposed developer API. This code base is the oldest of my plugin lineup being maintained and needs some major renovations (should probably expose futures in the API and utilize them for better database handling).